backend/data/scrapers/gened_scraper.py

```diff
@@ -0,0 +1,68 @@
+"""
+Program extracts raw data for Programs and place data in file
+'coursesPureRaw.json', ready for formatting.
+
+Step in the data's journey:
+    [ X ] Scrape raw data (gened_scraper.py)
+    [   ] Format scraped data
+    [   ] Customise formatted data
+"""
+
+import json
+
+import requests
+from data.scrapers.courses_scraper import TOTAL_COURSES
+from data.scrapers.payload import HEADERS, create_payload_gened
+from data.utility import data_helpers
+
+URL = "https://www.handbook.unsw.edu.au/api/es/search?appliedFilters="
+
+def scrape_gened_data(year=None):
+    """
+    Retrieves gen ed data for all undergraduate programs
+    Stores in genedPureRaw.json as list of course codes for each program code in programsFormattedRaw.json
+    """
+    faculty_list = []
+    gened_raw = {}
+    cl_id = ""
+    academicOrg = ""
+
+    programs_formatted = data_helpers.read_data("data/scrapers/programsFormattedRaw.json")
+    for program_code in programs_formatted:
+        program = programs_formatted[program_code]
+        faculty = program["faculty"]
+        if faculty in faculty_list:
+            continue
+        else:
+            faculty_list.append(faculty)
+
+        if faculty == "Faculty of Arts, Design and Architecture" or faculty == "Faculty of Law and Justice":
+            academicOrg = "academicOrg"
+        else:
+            academicOrg = "parentAcademicOrg"
+        cl_id = program[academicOrg]
+        r = requests.post(URL, data=json.dumps(
+            create_payload_gened(
+            TOTAL_COURSES,
+            "unsw_psubject",
+            cl_id,
+            academicOrg,
+            year
+            )
+        ), headers=HEADERS)
```

Comment:
Can we check this request for success / error and then print out to console or some log what course it failed for.
This can be done right here or, you can create a list of dicts containing data about failed gen_ed scrapes and then print that only at the very end

Reply:
pushed change but don't think it's the best/right way to do it